xblock/mixins.py

```diff
@@ -522,7 +522,7 @@ def _add_field(self, node, field_name, field):
             elem.text = value
             node.insert(0, elem)
         else:
-            node.set(field_name, value)
+            node.set(field_name, '' if value is None else value)
```

Comment:
can you add a comment in the code why this is necessary and also add a test for this in this repo?